ArduCopter/mode_auto.cpp

```diff
@@ -846,6 +851,11 @@ void ModeAuto::wp_run()
         }
     }
 
+    // if set and no pilot input for 2 sec weathervane copter into wind
+    if (allows_weathervaning()) {
+        auto_yaw.update_weathervane(target_yaw_rate, wp_nav->get_roll(), wp_nav->get_pitch());
```

Comment:
it's a bit of a drag that we have this yaw stuff creeping out into each controller and the input handling and decision of which yaw control method to use becomes fragmented.

Reply:
Yes, agreed this could be better.  The 'process pilots yaw input' code block above this sets the precident.  I think we can be tidier and move all of the yaw handling into autoyaw and then we simply call an 'update_yaw' function instead.  But that should be a seperate PR.